### ISSUE TITLE
Expected indentation of NaN spaces but found 6

### ISSUE THREAD

**Issue by gillez** [eslint/eslint#7756]
**Tell us about your environment**
MacOS v10.11.6
* **ESLint Version:**
v3.11.1
* **Node Version:**
v6.9.1
* **npm Version:**
v4.0.3

**What parser (default, Babel-ESLint, etc.) are you using?**
eslint from command line and VS Code ESLint plugin v1.1.0

**Please show your full configuration:**

```
{
  "env": {
    "browser": true,
    "es6": true,
    "node": true,
    "jquery": true
  },
  "globals": {
    "$": false,
    "angular": false,
    "listFunctions": false,
    "PouchDB": false,
    "yohelpers": false,
    "Property": false,
    "ActionData": false,
    "serverData": true
  },
  "extends": "eslint:recommended",
  "parserOptions": {
    "sourceType": "module"
  },
  "rules": {
    "indent": [
      "error",
      2,
      {
        "SwitchCase": 1,
        "CallExpression":{
          "arguments":"off"
        }
      }
    ],
    "linebreak-style": [
      "error",
      "unix"
    ],
    "quotes": [
      "off",
      "double"
    ],
    "semi": [
      "error",
      "always"
    ],
    "no-console": [
      // "warn"
      "off"
    ],
    "no-unused-vars": [
      "error",
      {
        "argsIgnorePattern": "^_unused_"
      }

    ]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
    async.whilst(
      function() { 
        return count < numRows; 
      },
      function(callback) { 
        /* Process next block... */
        callback();
      },
      function (err) {
        // All blocks completed.
        console.log("all blocks completed, record count", count);
        callback(err);
      }
    );
```

**What did you expect to happen?**
Pass with no warnings/errors (indent is set to 2)

**What actually happened? Please include the actual, raw output from ESLint.**

 160:7   error  Expected indentation of NaN spaces but found 6   indent
 163:7   error  Expected indentation of NaN spaces but found 6   indent
 167:7   error  Expected indentation of NaN spaces but found 6   indent

✖ 3 problems (3 errors, 0 warnings)

**Comment by gillez** [comment]
Note: the following layout removes the errors.  

    async.whilst(function() { 
      return count < numRows; 
    }, function(callback) { 
      /* Process next block... */
      callback();
    }, function (err) {
      // All blocks completed.
      console.log("all blocks completed, record count", count);
      callback(err);
    });

This layout is fine for me, but I still want to flag the message "Expected indentation of NaN spaces" as it clearly isn't correct.

**Comment by not-an-aardvark** [comment]
Thanks for the report. I noticed you're using ESLint 3.11.1; does this problem still occur when you upgrade to the latest version of ESLint?

**Comment by gillez** [comment]
Thanks for the reply.  I've just upgraded ESLint to 3.12.1 and still see the same issue.

**Comment by not-an-aardvark** [comment]
I think this is occurring because the `CallExpression.arguments` option does not currently support a value of `off`. The `CallExpression` option is ignored by default when no option is provided, so you can just omit that option to get the desired behavior.

**Comment by gillez** [comment]
Thanks for the explanation.  I have removed the CallExpression section and I no longer see those warnings.
As you have explained that the validation/errors can't be improved due to backwards compatibility, I am happy to close this issue.

**Comment by gillez** [comment]
The only thing I would add is that maybe it should be mentioned in the docs that "off" should not be used and the block should be ommitted instead (the docs do say that off is the default but not that it shouldn't be specified).
http://eslint.org/docs/rules/indent

**Comment by gillez** [comment]
Actually forget the last comment.  I've re-read the docs and it was my misunderstanding.  I thought that "off" is generally used to turn options off, but I think it's only for configuring rules, not for configuring options inside rules.  If that's the general behaviour, I don't think it needs to be explicitly mentioned for this case.

**Comment by platinumazure** [comment]
Unrelated to this issue, it'd be nice if we could support an explicit "off" in case an upstream configuration turns it on and the user wants it off. Is that in the works @not-an-aardvark?

**Comment by not-an-aardvark** [comment]
I think there was some discussion about it awhile ago, but we never ended up getting around to adding it. I agree that it would be nice if we could support it.
